src/frontend/src/binder/relation/generate_series.rs

```diff
@@ -59,6 +60,23 @@ impl Binder {
             .flatten_ok()
             .try_collect()?;
 
+        type_check(&exprs)?;
+
         Ok(BoundGenerateSeriesFunction { args: exprs })
     }
 }
+
+fn type_check(exprs: &Vec<ExprImpl>) -> Result<()> {
```

Comment:
It seems that I can't use expr eval dummy chunk for type detection on the front end like on the back end. Any suggestions? @xiangjinwu

Reply:
You did it right - just check `expr.return_type`.

* Also suggest to return a `DataType` from this function and store it in `BoundGenerateSeriesFunction`, to be used by bind context (line 48 above) and schema (`plan_generate_series_function` line 95).

* `BoundGenerateSeriesFunction` can actually store `start`, `stop`, `step` directly rather than `args: Vec`, so that we do not need to match args twice (binder & planner).
  * Unless we want to generalize it from `BoundGenerateSeriesFunction` to `BoundTableFunction`.

Reply:
> You did it right - just check expr.return_type.

```
    let Some((ExprImpl::Literal(start), ExprImpl::Literal(stop),ExprImpl::Literal(step))) = exprs.next_tuple() else {
        return Err(ErrorCode::BindError("Invalid arguments for Generate series function".to_string()).into())
    };
```
I've run into a problem where writing it this way makes it impossible if I want the function to be able to accept expressions, like this  `SELECT * FROM generate_series('2'::INT + '2'::INT,,'10'::INT,'2'::INT)`

Reply:
> * Also suggest to return a `DataType` from this function and store it in `BoundGenerateSeriesFunction`, to be used by bind context (line 48 above) and schema (`plan_generate_series_function` line 95).

great suggestion!

Reply:
My bad ... I meant the `match (start.return_type(), stop.return_type(), step.return_type())` part is right.

As for the first part - just skip matching against `Literal`. We can even `start = exprs[0]` and call `start.return_type()`.